mwdb/web/src/components/RichAttribute/RichAttributePreview.js

```diff
@@ -80,20 +81,13 @@ export default function RichAttributePreview({
         templateState.chosenExample !== "custom"
             ? exampleTemplates[templateState.chosenExample].value
             : templateState.valueInput;
-    let renderedValue,
-        contextValue,
-        invalid = false;
+    let contextValue;
     try {
         contextValue = makeContext(JSON.parse(value));
-        renderedValue = renderValue(template, contextValue, {
-            searchEndpoint: "/",
-        });
     } catch (e) {
-        renderedValue = e.toString();
         contextValue = null;
-        invalid = true;
+        setInvalid(true);
     }
```

Comment:
I had infinite render loop due to `setInvalid(true)` triggering re-renders when value is empty. I would wrap it in useEffect and move contextValue to state.

Reply:
moved contextValue to state and wrap it in useEffect hook to prevent re-renders, but I have to leave contextValue conversion in RichAttributeRenderer.js to convert value with renderValue function